### PR TITLE
feat(Toggle): allow react node for labels

### PR DESCRIPTION
**Pull request by xendke**
### Overview

Update the label prop type to accept ReactNode.

### PR Checklist

- [x] Related to designs: [Figma](https://www.figma.com/file/m47AgRC46zfFVQe5vcuv6R/Checkout-enhancements-post-first-experiement?node-id=770%3A74076)
- [x] Related to JIRA ticket: [GRW-325](https://codecademy.atlassian.net/browse/GRW-325)
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [x] This PR includes testing instructions tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

Open Storybook preview, go to Toggle section, observe the new example using `<Text>` for label

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Portal       | [Portal](https://github.com/codecademy-engineering/mono/pull/499)  | [PR Env](https://tengu.codecademy.com/checkout?PR_ENV=portal-app-pr-499)  |
| Monolith       | [Monolith](https://github.com/codecademy-engineering/Codecademy/pull/33300)  | [PR Env](https://pr-33300-monolith.dev-eks.codecademy.com/)  |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.220-alpha.342d6d.0<br/>@codecademy/gamut-labs@33.6.11-alpha.342d6d.0<br/>@codecademy/gamut@51.1.1-alpha.342d6d.0<br/>@codecademy/monaco@0.6.98-alpha.342d6d.0<br/>@codecademy/styleguide@62.0.5-alpha.342d6d.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.220-alpha.51a5bf.0<br/>@codecademy/gamut-labs@33.6.11-alpha.51a5bf.0<br/>@codecademy/gamut@51.1.1-alpha.51a5bf.0<br/>@codecademy/monaco@0.6.98-alpha.51a5bf.0<br/>@codecademy/styleguide@62.0.5-alpha.51a5bf.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.220-alpha.ff1dce.0<br/>@codecademy/gamut-labs@33.6.11-alpha.ff1dce.0<br/>@codecademy/gamut@51.1.1-alpha.ff1dce.0<br/>@codecademy/monaco@0.6.98-alpha.ff1dce.0<br/>@codecademy/styleguide@62.0.5-alpha.ff1dce.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://635fd3da6d441542b75ae3be--gamut-preview.netlify.app>https://635fd3da6d441542b75ae3be--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/635fd3da6d441542b75ae3be>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.220-alpha.ba4a46.0<br/>@codecademy/gamut-labs@33.6.11-alpha.ba4a46.0<br/>@codecademy/gamut@51.1.1-alpha.ba4a46.0<br/>@codecademy/monaco@0.6.98-alpha.ba4a46.0<br/>@codecademy/styleguide@62.0.5-alpha.ba4a46.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://635fe94a6d44155c685ae51b--gamut-preview.netlify.app>https://635fe94a6d44155c685ae51b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/635fe94a6d44155c685ae51b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.220-alpha.d84711.0<br/>@codecademy/gamut-labs@33.6.11-alpha.d84711.0<br/>@codecademy/gamut@51.1.1-alpha.d84711.0<br/>@codecademy/monaco@0.6.98-alpha.d84711.0<br/>@codecademy/styleguide@62.0.5-alpha.d84711.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://635fee1d1beb545ede00e8e8--gamut-preview.netlify.app>https://635fee1d1beb545ede00e8e8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/635fee1d1beb545ede00e8e8>Deploy Logs</a></p>

File: packages/gamut/src/Toggle/utils.tsx [hunk 1]
```diff
@@ -12,11 +12,12 @@ export const getToggleElementProps = <Props extends ToggleProps>({
   Props,
   'ariaLabel' | 'as' | 'checked' | 'disabled' | 'label' | 'onChange' | 'onClick'
 >): Omit<ToggleInputStyledProps, 'size' | 'color'> => {
+  const id = label && typeof label === 'string' ? label : ariaLabel;
```
Comment (jakemhiller) [inline packages/gamut/src/Toggle/utils.tsx:15]: `ariaLabel` should be required if `label` is a `ReactNode`. Could you update the `ToggleLabelProps` union to include that?

**Review by jakemhiller — DISMISSED** [review]
Update the `ToggleLabelProps` union to prevent `ReactNode` labels without `ariaLabel` prop

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.220-alpha.a6be5b.0<br/>@codecademy/gamut-labs@33.6.11-alpha.a6be5b.0<br/>@codecademy/gamut@51.1.1-alpha.a6be5b.0<br/>@codecademy/monaco@0.6.98-alpha.a6be5b.0<br/>@codecademy/styleguide@62.0.5-alpha.a6be5b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://63601bc16c01dd1ccdcea729--gamut-preview.netlify.app>https://63601bc16c01dd1ccdcea729--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/63601bc16c01dd1ccdcea729>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]
lgtm + discriminated union is ✨

**Pull request merged**